org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/IFilterOutput.java

```diff
@@ -53,7 +55,42 @@ public interface IFilterOutput {
 	 * @param newTargets
 	 *            new targets of branches
 	 */
+	void replaceBranches(AbstractInsnNode source,
+			Map<AbstractInsnNode, Collection<InstructionBranch>> newTargets);
+
+	/**
+	 * @param source
+	 *            instruction which branches should be replaced
+	 * @param newTargets
+	 *            new targets of branches
+	 * @deprecated use {@link #replaceBranches(AbstractInsnNode, Map)} instead
+	 */
+	@Deprecated
 	void replaceBranches(AbstractInsnNode source,
 			Set<AbstractInsnNode> newTargets);
 
+	/**
+	 * {@link #instruction} and index of its {@link #branch}.
```

Comment:
It will be nice to add a comment about branch indexes for `JumpInsnNode`, `TableSwitchInsnNode`, `LookupSwitchInsnNode` and others.